### PR TITLE
add scale to zero status in non serverless applications

### PR DESCRIPTION
**Pull request by karthikjeeyar**
Show scale to zero on a nonserverless application in topology

screencast-localhost-9000-2019 07 16-13-49-27

Story: https://jira.coreos.com/browse/ODC-1151

### CODE REVIEW

**Comment by christianvogt** [comment]
@karthikjeeyar I see nothing here related to the idle state.

**Comment by karthikjeeyar** [comment]
@christianvogt used the annotation to check the idle state

**Comment by karthikjeeyar** [comment]
@sspeiche I have used `oc idle <dc-name>` to idle the application, which adds the following annotation `idling.alpha.openshift.io/idled-at` to the DC, but the process of scaling up the resource back again is not removing the annotation. Is there any way to find if the resource being scaled up from idle state? any thoughts?
cc: @christianvogt

File: frontend/packages/dev-console/src/components/topology/topology-utils.ts [hunk 1]
```diff
@@ -292,11 +292,20 @@ export class TransformTopologyData {
     return !!deploymentsLabels['serving.knative.dev/configuration'];
   }
 
+  /**
+   * check if config is non serverless application.
+   * @param deploymentConfig
+   */
+  private isNonServerlessApplication(deploymentConfig: ResourceProps): boolean {
+    const annotations = _.get(deploymentConfig, 'metadata.annotations') || {};
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/topology-utils.ts]:
```suggestion
    return !!_.get(deploymentConfig, ['metadata.annotations', 'idling.alpha.openshift.io/idled-at']);
```

File: frontend/packages/dev-console/src/components/topology/topology-utils.ts [hunk 2]
```diff
@@ -292,11 +292,20 @@ export class TransformTopologyData {
     return !!deploymentsLabels['serving.knative.dev/configuration'];
   }
 
+  /**
+   * check if config is non serverless application.
+   * @param deploymentConfig
+   */
+  private isNonServerlessApplication(deploymentConfig: ResourceProps): boolean {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/topology-utils.ts]: This is a bad name for checking if the DC is idling. `isIdled` ?

**Comment by sspeiche** [comment]
> @sspeiche I have used `oc idle <dc-name>` to idle the application, which adds the following annotation `idling.alpha.openshift.io/idled-at` to the DC, but the process of scaling up the resource back again is not removing the annotation. Is there any way to find if the resource being scaled up from idle state? any thoughts?
> **Update**: Autoscaling removes the `idling.alpha.openshift.io/idled-at` annotations as expected, but if the user manually scale up the application from UI, then the annotation is not getting removed.
> 
> cc: @christianvogt

Is this true for 3.11 as well? How did the 3.11 console handle it?

**Comment by karthikjeeyar** [comment]
@christianvogt resolved the conflicting file and pushed the code.

**Comment by karthikjeeyar** [comment]
> > @sspeiche I have used `oc idle <dc-name>` to idle the application, which adds the following annotation `idling.alpha.openshift.io/idled-at` to the DC, but the process of scaling up the resource back again is not removing the annotation. Is there any way to find if the resource being scaled up from idle state? any thoughts?
> > **Update**: Autoscaling removes the `idling.alpha.openshift.io/idled-at` annotations as expected, but if the user manually scale up the application from UI, then the annotation is not getting removed.
> > cc: @christianvogt
> 
> Is this true for 3.11 as well? How did the 3.11 console handle it?

Yes, it is happening in 3.11 as well. It looks like only upon the network request on the idle applications are getting scaled up and annotations are removed. 3.11 console handles based on the replica count and annotation condition, we are also checking the same. I am bringing this up because it might lead to confusion in the corner case where if `user` manually idles and unIdles the application from UI.
related bug : (https://bugzilla.redhat.com/show_bug.cgi?id=1370434)

**Comment by divyanshiGupta** [comment]
> > > @sspeiche I have used `oc idle <dc-name>` to idle the application, which adds the following annotation `idling.alpha.openshift.io/idled-at` to the DC, but the process of scaling up the resource back again is not removing the annotation. Is there any way to find if the resource being scaled up from idle state? any thoughts?
> > > **Update**: Autoscaling removes the `idling.alpha.openshift.io/idled-at` annotations as expected, but if the user manually scale up the application from UI, then the annotation is not getting removed.
> > > cc: @christianvogt
> > 
> > 
> > Is this true for 3.11 as well? How did the 3.11 console handle it?
> 
> Yes, it is happening in 3.11 as well. It looks like only upon the network request on the idle applications are getting scaled up and annotations are removed. 3.11 console handles based on the replica count and annotation condition, we are also checking the same. I am bringing this up because it might lead to confusion in the corner case where if `user` manually idles and unIdles the application from UI.
> related bug : (https://bugzilla.redhat.com/show_bug.cgi?id=1370434)

@karthikjeeyar I agree with you. It will also add inconsistency. We should let the backend guys know about this issue if they are not aware of it.

@karthikjeeyar the code looks good to me and is working fine for me.

/lgtm

**Pull request merged**